recommenders/models/sar/sar_singlenode.py

```diff
@@ -267,6 +268,8 @@ def fit(self, df):
         # free up some space
         del temp_df
 
+        # creates an array with the frequency of every unique user/item
+        self.user_frequencies = np.count_nonzero(self.user_affinity.toarray(), axis=1)
```

Comment:
This is an inefficient way to count these, because it converts a sparse csr matrix to a dense array.

Reply:
Another thing is that it would be much better if this computation were "on demand" i.e. it should not be computed inside `fit()` since it is not going to be used in production or in most other settings.

Reply:
updated with the latest commit. does that work?